mimir-build-image/Dockerfile

```diff
@@ -4,8 +4,8 @@
 # Provenance-includes-copyright: The Cortex Authors.
 
 FROM k8s.gcr.io/kustomize/kustomize:v4.5.5 as kustomize
-FROM alpine/helm:3.8.2 as helm
-FROM golang:1.19.3-bullseye
+FROM alpine/helm:3.11.1 as helm
```

Comment:
We only copy Helm from this image (`COPY --from=helm /usr/bin/helm /usr/bin/helm` instruction). Do we need to update this image in this PR?

Reply:
We don't need to upgrade Helm in this PR. I just suggested trying to upgrade it, after Krajo said an upgrade should be safe. At the beginning it wasn't clear if the build was failing due to Helm (the error was difficult to understand), so it was worth trying.

Reply:
@pstibrany my reply was basically to confirm that it's not necessary in this PR :)